svc-bie-kafka/src/main/java/gov/va/vro/services/bie/utils/BieMessageUtils.java

```diff
@@ -1,65 +1,88 @@
+import gov.va.vro.model.biekafka.ContentionEvent;
+import gov.va.vro.model.biekafka.annotation.Ignore;
+import gov.va.vro.model.biekafka.annotation.TargetEvents;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.avro.generic.GenericRecord;
 import org.springframework.util.StringUtils;
 
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
-import java.util.HashSet;
+import java.util.Arrays;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 @Slf4j
 public class BieMessageUtils {
-  private static final Set<String> IGNORED_FIELDS = new HashSet<>();
-
-  static {
-    // Add the names of fields to be ignored
-    IGNORED_FIELDS.add("status");
-    IGNORED_FIELDS.add("statusMessage");
-    IGNORED_FIELDS.add("eventType");
-    IGNORED_FIELDS.add("notifiedAt");
-    // description field disabled for now until BIE added and deployed the new kafka schema in dev
-    IGNORED_FIELDS.add("description");
-  }
-
-  public static BieMessagePayload processBieMessagePayloadFields(GenericRecord genericRecord) {
+  public static BieMessagePayload processBieMessagePayloadFields(
+      ContentionEvent contentionEvent, GenericRecord genericRecord) {
     BieMessagePayload payload = BieMessagePayload.builder().status(200).build();
 
     for (Field field : BieMessagePayload.class.getDeclaredFields()) {
       String fieldName = field.getName();
 
-      // Skip the field if it's in the ignored list
-      if (IGNORED_FIELDS.contains(fieldName)) {
+      // Skip the field if it has @Ignore annotation or is null
+      if (field.isAnnotationPresent(Ignore.class)) {
         continue;
       }
 
-      String capitalizedFieldName = StringUtils.capitalize(fieldName);
-      Object value = genericRecord.get(capitalizedFieldName);
+      if (field.isAnnotationPresent(TargetEvents.class)) {
+        String[] annotationValues = field.getAnnotation(TargetEvents.class).value();
 
-      if (value == null) {
-        continue; // Skip setting the field if the value is null
-      }
+        if (!isValidTopicNames(annotationValues)) {
+          throw new IllegalArgumentException(
+              "Invalid topic names in annotation for field: " + fieldName);
+        }
 
-      try {
-        String setterMethodName = "set" + capitalizedFieldName;
-        Method setterMethod = BieMessagePayload.class.getMethod(setterMethodName, field.getType());
-
-        if (field.getType().isAssignableFrom(value.getClass())) {
-          setterMethod.invoke(payload, value);
-        } else {
-          log.warn(
-              "Type mismatch for field '{}'. Expected type: '{}', Actual value: '{}'",
-              fieldName,
-              field.getType().getSimpleName(),
-              value);
+        boolean matchedTopicName =
+            Arrays.stream(annotationValues).anyMatch(v -> v.equals(contentionEvent.getTopicName()));
+
+        if (matchedTopicName) {
+          invokeSetterMethod(payload, field, genericRecord);
```

Comment:
Is the if statement still needed? It seems like the if else is calling the same method and everything else is inconsequential/for logging.

Reply:
Ah that makes sense. LGTM then